obs/stats.go

```diff
@@ -320,11 +320,13 @@ func (r StatsTraceReporter) ConsumeEvent(evt rcmgr.TraceEvt) {
 		tags := []tag.Mutator{tag.Upsert(scopeTag, scopeName), tag.Upsert(resourceTag, resource)}
```

Comment:
Does it make sense to allocate a slice of length 3 here, so the `append` later doesn't cause any allocs?

Reply:
I'm actually not appending to this slice. I make another slice of tags in case the event has both delta in/out (I don't think we are doing this anywhere but we can't assume we will never do this).

Reply:
I allocated the other slices upfront so maybe they'll be on the stack and not cause a heap alloc. Based on this SO answer/related tweet it seems like the Go 1.17 runtime (and above?) will do stack allocations here. 

I couldn't find any official documentation around this though.